libmproxy/web/app.py

```diff
@@ -239,6 +294,32 @@ def put(self):
         )
 
 
+class PluginList(RequestHandler):
+    def get(self):
+        def _flatten(plugin_list):
+            ret_arr = []
+            for plugin_type, plugin_dicts in dict(plugin_list).items():
+                for plugin_id, plugin_dict in plugin_dicts.items():
+                    new_dict = plugin_dict.copy()
+                    new_dict['id'] = plugin_id
+                    new_dict['type'] = plugin_type
+
+                    def _check_dict_for_callables(_dict):
+                        for k, v in _dict.items():
+                            if callable(v):
+                                _dict[k] = repr(v)
+                            if isinstance(v, dict):
+                                _check_dict_for_callables(v)
+
+                    _check_dict_for_callables(new_dict)
+                    ret_arr.append(new_dict)
+            return ret_arr
+
+        self.write(dict(
+            data=list(_flatten(self.master.plugins))
+        ))
```

Comment:
Following the parts above, let's just expose the list of content views in the Settings API endpoint?
(Exposing application state is surely something that's not ideal yet, but that's a different issue)